pennylane/tape/tapes/rewind.py

```diff
@@ -0,0 +1,182 @@
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Quantum tape for implementing the gradient method outlined in https://arxiv.org/abs/2009.02823,
+referred to here as the "rewind" method.
+"""
+import numpy as np
+
+# pylint: disable=protected-access
+import pennylane as qml
+
+from .jacobian_tape import JacobianTape
+
+
+class RewindTape(JacobianTape):
+    r"""Quantum tape for computing gradients using the `rewind <https://arxiv.org/abs/2009.02823>`__
+    method.
+
+    After a forward pass, the circuit is "rewound" by iteratively applying the inverse gate to scan
+    backwards through the circuit. This method is similar to the reversible method, but has a lower
+    time overhead and a similar memory overhead.
+
+    .. note::
+
+        The rewind analytic differentation method has the following restrictions:
+
+        * As it requires knowledge of the statevector, only statevector simulator devices can be
+          used.
+
+        * Only expectation values are supported as measurements.
+
+    This class extends the :class:`~.jacobian` method of the quantum tape to support analytic
+    gradients of qubit operations using the rewind method of analytic differentiation.
+    This gradient method returns *exact* gradients, however requires use of a statevector simulator.
+    Simply create the tape, and then call the Jacobian method:
+
+    >>> tape.jacobian(dev)
+
+    For more details on the quantum tape, please see :class:`~.JacobianTape`.
+    """
+
+    def jacobian(self, device, params=None, **options):
+        # The rewind tape only support differentiating expectation values of observables for now.
+        for m in self.measurements:
+            if m.return_type is not qml.operation.Expectation:
+                raise qml.QuantumFunctionError(
+                    f"The {m.return_type.value} return type is not supported with the rewind "
+                    f"gradient method"
+                )
+
+        method = options.get("method", "analytic")
+
+        if method == "device":
+            # Using device mode; simply query the device for the Jacobian
+            return self.device_pd(device, params=params, **options)
+        if method == "numeric":
+            return super().jacobian(device, params=params, **options)
+
+        supported_device = hasattr(device, "_apply_operation")
+        supported_device = supported_device and hasattr(device, "_apply_unitary")
+        supported_device = supported_device and device.capabilities().get("returns_state")
+
+        if not supported_device:
+            raise qml.QuantumFunctionError(
+                "The rewind gradient method is only supported on statevector-based devices"
```

Comment:
The device could be a statevector device, yet just not use the same internal functions as `default.qubit`, the `_apply_operation` and `_apply_unitary`.  

Maybe we should have one error for whether or not its a statevector device, and another for the internal functions.  This would allow the error message to be more precise.